server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java

```diff
@@ -191,6 +191,8 @@ public ManagedUser save(@PathVariable("userName") String userName, @RequestBody
         }
         logger.info("Saving {}", user);
 
+        user.setPassword(pwdEncode(user.getPassword()));
```

Comment:
I'm not quite understand why it need reset the password. Could you add a description?

Reply:
Debugging found that the password of the ‘user’ passed in by this method is not encoded. If the password is directly updated to the metadata, the authentication of the user name and password will fail  for bad credential.

Reply:
I see